### ISSUE TITLE
MSI for CLI should ensure it cleans up previous installs properly

### ISSUE THREAD

**Issue by billwert** [dotnet/cli#1355]
On one of my machines, I had run the MSI in November or December. I ran the MSI pointed to from dotnet.github.io yesterday, and was getting an error running 'dotnet restore' on anything. The actual cause of the error was that Microsoft.Extensions.PlatformAbstractions.dll was an old version in %ProgramFiles%\dotnet\bin, and was missing a method. Completely uninstalling the CLI through Add/Remove Programs (somehow I had 3 entries?) and reinstalling fixed the problem.

The MSI should ensure there is no torn state.

**Comment by robmen** [comment]
The fact that you had 3 entries suggests that is the root problem. If you reproduce the issue and capture [Windows Installer log files](https://support.firegiant.com/entries/24024218-Create-a-log-file-) it should be possible to diagnose exactly what happened.

Going forward the code does appear to now upgrade correctly.

**Comment by Sridhar-MS** [comment]
@billwert MSIs were made to do build to build upgrades very recently -  https://github.com/dotnet/cli/pull/836

This is mainly because we did not have proper versioning in MSI in the older builds. This is fixed now.

**Comment by GuardRex** [comment]
@Sridhar-MS I may have just seen this with the new installer this morning. I installed using the new MSI this morning (`-001290-beta`), and it didn't remove (or upgrade) the prior tools ...

![capture](https://cloud.githubusercontent.com/assets/1622880/12988285/dbb8c048-d0b4-11e5-9f92-fbfafb479ee5.PNG)

At that point if you uninstall prior tooling, it removes `dotnet` from the PATH, so you really end up uninstalling both and re-installing the latest MSI.

**Comment by Sridhar-MS** [comment]
@GuardRex This seems like a real issue. I will take a look at this.

**Comment by robmen** [comment]
The `dev` build there has a version of `1.0.1.001200` but the `beta` build has a version of `1.0.0.001290`. Seems like the `beta` build should have blocked since it has a lower version.

@GuardRex in your %TMP% folder you should see log files (for the new dotnet installs), can you attach those?

**Comment by GuardRex** [comment]
@robmen Sure ... let me see if I can find those for you. brb

**Comment by Petermarcu** [comment]
Yeah, this likely spanned the version breaking change that was made. Dont plan on doing it again.

**Comment by GuardRex** [comment]
@robmen There are four of them here ... how do you want me to get them to you?

![capture3](https://cloud.githubusercontent.com/assets/1622880/12989152/dcfe8128-d0b8-11e5-864e-a8b989a3997d.PNG)

**Comment by robmen** [comment]
Yeah, probably easiest to .zip them up. Do you have the log files for the `dev` build? @Petermarcu suggests this is part of a breaking version change.

**Comment by GuardRex** [comment]
No, I don't see them. I just see these four files. I can zip, but how do I post them to you?

**Comment by robmen** [comment]
You can just attach them to the issue here. However, without log files for `dev` might not be able to get full story. When did you install `dev`?

**Comment by GuardRex** [comment]
Ah ... good question ... um ... The MSI was not updated for a while ... then it was offline for a while. I think it was a full week ago or so. It was several days ago for sure.

[files.zip](https://github.com/dotnet/cli/files/127073/files.zip)

**Comment by GuardRex** [comment]
I trashed the installer I used for `-001200-dev`, too, so I won't be able to repro this here I don't think.

**Comment by robmen** [comment]
My hypothesis was correct:

    [0420:2254][2016-02-11T10:09:48]i201: Planned package: dotnet_win_x64.1.0.0.001290.msi, state: Obsolete, default requested: None, ba requested: None, execute: None, rollback: None, cache: No, uncache: No, dependency: None

In the future, the bundle should detect a newer version of itself and block. In this case, this is pre-bundle install and since this only happens due to the version breaking change, I'm inclined to suggest not adding more code in the bundle to try to detect this state and block again.

Basically, yes, this is a bug due to the version change and everything should work better in the future! :smile_cat:

**Comment by Petermarcu** [comment]
Yeah, that's how we thought about it. Moving forward things should be fine and we didn't want to add more logic for this point in time problem.

**Comment by blackdwarf** [comment]
@robmen @billwert should we then close this issue?

**Comment by robmen** [comment]
Based on @Petermarcu's above confirmation, yes.

**Comment by GuardRex** [comment]
There have been no further issues here over-installing with new MSI's as they come out.

**Comment by blackdwarf** [comment]
@robmen @GuardRex thanks guys, closing.

@billwert if you see this reproing again, reopen.
